lib/domain/qr_code/provider/scanner_provider.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:superformula_test/core/service_locator.dart';
+import 'package:superformula_test/domain/qr_code/model/validator_model.dart';
+import 'package:superformula_test/domain/qr_code/repository/qr_code_repo.dart';
+
+class ScannerProvider extends ChangeNotifier {
+  final RestClient client = ServiceLocator.locator<RestClient>();
+
+  ScannerProvider();
+
+  String? result = '';
+  String? format = '';
+
+  ///computes the values of the Barcode data and assign them accordingly
+  Future<void> resolveDataScan(Barcode data) {
+    format = describeEnum(data.format);
+    result = data.code;
+    notifyListeners();
+    return Future.value();
+  }
+
+  void clearData() {
+    format = null;
+    result = null;
+    notifyListeners();
+  }
+
+//posts the value of the scanned QR to the endpoint
+  Future<bool> submitQR() async {
```

Comment:
question: Why is this returning a bool? It seems to be only for a single unit test.

Reply:
In the WIP simulation, I wanted a boolean to be returned in a try block or the `.then`, and in the error case or catch block, I return false. Based on that boolean I would then do an update on the page in a real scenario.